python/ecosystem.go

```diff
@@ -0,0 +1,73 @@
+package python
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/internal/indexer"
+)
+
+var scanners = []indexer.PackageScanner{&Scanner{}}
+
+// NewEcosystem provides the set of scanners for the python ecosystem.
+func NewEcosystem(ctx context.Context) *indexer.Ecosystem {
+	return &indexer.Ecosystem{
+		PackageScanners:      func(_ context.Context) ([]indexer.PackageScanner, error) { return scanners, nil },
+		DistributionScanners: func(_ context.Context) ([]indexer.DistributionScanner, error) { return nil, nil },
+		RepositoryScanners:   func(_ context.Context) ([]indexer.RepositoryScanner, error) { return nil, nil },
+		Coalescer:            NewCoalescer,
+	}
+}
+
+func NewCoalescer(_ context.Context) (indexer.Coalescer, error) {
+	return &coalescer{}, nil
+}
+
+type coalescer struct {
+}
+
+func (c *coalescer) Coalesce(ctx context.Context, ls []*indexer.LayerArtifacts) (*claircore.IndexReport, error) {
+	repos := make(map[string]string)
```

Comment:
Can you comment here explaining your Coalescing strategy ?

We must evaluate the "newest" pip database (the ones higher in the layer array) to get the final set of pip packages in the resulting image.

Reply:
I'm not sure what you mean by "pip database".

The data model currently doesn't have any concept of operations on the set of packages, it just presents the complete set on each layer. Because we can't do that, this does the conservative thing and assumes nothing is uninstalled.

Reply:
discussed out of band, we'll determine the best way to go  about this in subsequent PR.